### ISSUE TITLE
Did you try to use multigpu for training?

### ISSUE THREAD

**Issue by K-Mike** [UKPLab/sentence-transformers#43]

**Comment by nreimers** [comment]
Haven't tried it yet.

**Comment by LeenaShekhar** [comment]
Hi Nils,

Could you please tell me how did you use DataParallel to utilize multigpu? When I try to wrap SentenceTransformer, then I am getting CUDA OOM error even though the model training works just fine on 1 gpu.

**Comment by nreimers** [comment]
In the models.Transformer class, you can wrap the auto_model in DataParallel(AutoModel...)

**Comment by LeenaShekhar** [comment]
Hi Nils.

I am wondering if just adding this actually worked for you or other changes were needed as well? I wrapped the Auto model in DataParallel and used .module to refer to it. Wondering if input data also has to be put on device? 

After using the Asymmetric model (with 2 transformer large models as word embedding model), the model goes OOM even with batch size 2 where the sense layer output is 32. Hence I was thinking may be thinking about multi-gpu support is kind of important.

**Comment by nreimers** [comment]
Hi @LeenaShekhar 
My experiments with wrapping the model in DataParallel was rather unsuccessful: Training on two GPUs was slower than training on a single GPU.

Yes, if you use two transformer large models, it would make sense to have them on two separate GPUs. However, this would require quite a lot of changes in the code, as your data has to go to the right GPU, and the results have to come back to one GPU where you do the final loss computation.

**Comment by LeenaShekhar** [comment]
Yes, you are right. I was thinking initially data will be on cuda:0 then the left part of the network can be on cuda:0 while the right on cuda:1 and then finally the output on cuda:0. It would need many changes like you said and am not sure how the impact will be on speed due to so much shuffling tensors around. I will try it to see if the model fits into the memory though. 

Thank you.
